### ISSUE TITLE
Error when using Advanced queries with ":" in key

### ISSUE THREAD

**Issue by romanziske** [ajclopez/mongo-spring-search#2]
When parsing following query:
`filter=(captures.core:frequency>1 AND captures.core:frequency<5)`

This log appears: 
`line 1:1 no viable alternative at input 'captures.core:frequency'`

The resulted mogog query is:
`Mongo Query: {}`

**Comment by ajclopez** [comment]
Hi @romanziske, 

Thanks for reporting the issue. Can you give me an example of your database?

**Comment by romanziske** [comment]
Hi, 

typically, the stored documents look like this:

```
{
   "global": {
      "core:datatype": "cu8",
      "core:version": "1.0.0",
      "core:dataset": "non-conforming-dataset-01.dat"
   },
   "captures": [
      {
         "core:sample_start": 0,
         "core:header_bytes": 4,
         "core:frequency": 354
      },
      {
         "core:sample_start": 500,
         "core:header_bytes": 4,
         "core:frequency": 100,
      }
   ],
   "annotations": []
}
```

**Comment by romanziske** [comment]
Hi, 

typically, the stored documents look like this:

```
{
   "global": {
      "core:datatype": "cu8",
      "core:version": "1.0.0",
      "core:dataset": "non-conforming-dataset-01.dat"
   },
   "captures": [
      {
         "core:sample_start": 0,
         "core:header_bytes": 4,
         "core:frequency": 354
      },
      {
         "core:sample_start": 500,
         "core:header_bytes": 4,
         "core:frequency": 100,
      }
   ],
   "annotations": []
}
```

**Comment by ajclopez** [comment]
Hi @romanziske,

I have released a new version [1.0.4](https://github.com/ajclopez/mongo-spring-search/releases/tag/1.0.4) to support ":" character in key name.

Let me know your comments.

Thank you for reporting the issue.
